### PR TITLE
8328673: Convert closed text/html/CSS manual applet test to main

### PR DESCRIPTION
**Pull request by prsadhuk**
Couple closed manual applet text/html/CSS tests are converted to main based and opensourced

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328673](https://bugs.openjdk.org/browse/JDK-8328673): Convert closed text/html/CSS manual applet test to main (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [8ff80bbb](https://git.openjdk.org/jdk/pull/18424/files/8ff80bbbc1199d2ad87df1d601417c20b425a02f)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18424/head:pull/18424` \
`$ git checkout pull/18424`

Update a local copy of the PR: \
`$ git checkout pull/18424` \
`$ git pull https://git.openjdk.org/jdk.git pull/18424/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18424`

View PR using the GUI difftool: \
`$ git pr show -t 18424`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18424.diff">https://git.openjdk.org/jdk/pull/18424.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18424#issuecomment-2011694480)

### CODE REVIEW

File: test/jdk/javax/swing/text/html/CSS/bug4271058.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4270889 4271058 4285098
+ * @summary Tests that <table border>, <table align> and <td width> tags work
+ * @library /open/test/jdk/java/awt/regtesthelpers
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/CSS/bug4271058.java:28]: Need to remove "/open/test/jdk" here and in other test also.

**Review by kumarabhi006 — APPROVED** [review]
Please update the library path.

File: test/jdk/javax/swing/text/html/CSS/bug4271058.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/text/html/CSS/bug4271058.java]: Good catch...forgot to modify while copying...

File: test/jdk/javax/swing/text/html/CSS/bug4271058.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        """;
+
+    public static void main(String[] args) throws Exception {
+         PassFailJFrame.builder()
+                .title("CSS tag Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(15)
+                .columns(30)
+                .testUI(bug4271058::createTestUI)
+                .build()
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/text/html/CSS/bug4271058.java:60]: I guess the test will benefit from a screenshot feature to capture how the table is rendered.
```suggestion
                .testUI(bug4271058::createTestUI)
                .screenCapture()
                .build()
```

The feature to take a screenshot automatically, when Fail button is pressed, hasn't been implemented yet. It's tracked under [JDK-8317114](https://bugs.openjdk.org/browse/JDK-8317114).

File: test/jdk/javax/swing/text/html/CSS/bug4286458.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        String text =
+            "</html><body><table border=\"1\" cellpadding=\"-10\">" +
+            "<tr><td>This line should be clearly readable</td></tr>" +
+            "</table></body></html>";
+
+        JFrame f = new JFrame("bug4286458");
+        JEditorPane jep = new JEditorPane();
+        jep.setEditorKit(new HTMLEditorKit());
+        jep.setEditable(false);
+        jep.setText(text);
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/text/html/CSS/bug4286458.java:65]:
```suggestion
        JEditorPane jep = new JEditorPane("text/html", text);
        jep.setEditable(false);
```
Could be reduced to.

File: test/jdk/javax/swing/text/html/CSS/bug4286458.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                .columns(30)
+                .testUI(bug4286458::createTestUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JFrame createTestUI() {
+
+        String text =
+            "</html><body><table border=\"1\" cellpadding=\"-10\">" +
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/text/html/CSS/bug4286458.java:57]:
```suggestion
            "<html><body><table border=\"1\" cellpadding=\"-10\">" +
```
It should start with an opening tag.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/text/html/CSS/bug4271058.java (hunk 2 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/text/html/CSS/bug4271058.java]: ok

File: test/jdk/javax/swing/text/html/CSS/bug4286458.java (hunk 3 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/text/html/CSS/bug4286458.java]: ok

File: test/jdk/javax/swing/text/html/CSS/bug4286458.java (hunk 4 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/text/html/CSS/bug4286458.java]: yes

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**